src/db/models/OffCampusOfferModel.ts

```diff
@@ -82,4 +86,39 @@ export class OffCampusOfferModel extends Model<OffCampusOfferModel> {
     defaultValue: OfferStatusEnum.ONGOING,
   })
   status: OfferStatusEnum;
+
+  @AfterBulkCreate
+  static async sendEmailHook(instance: OffCampusOfferModel[]) {
+    console.log("New entries created");
+    const mailerService = new MailerService();
+
+    // Iterate over each newly created instance
+    for (const offer of instance) {
+      // Find the student associated with this offer
+      const student = await StudentModel.findByPk(offer.studentId);
+      if (!student) {
+        console.error(`Student not found for offer ${offer.id}`);
+        continue; // Skip to the next offer if student not found
+      }
+
+      // Find the user associated with this student
+      const user = await UserModel.findByPk(student.userId);
+      if (!user) {
+        console.error(`User not found for student ${student.id}`);
+        continue; // Skip to the next offer if user not found
+      }
+
+      // Prepare the email data
+      const dto: SendEmailDto = {
+        from: { name: "TPC Portal", address: "aryangkulkarni@gmail.com" },
+        // recepients: [{ address: "me210003016@iiti.ac.in" }],
+        recepients: [{ address: user.email }],
+        subject: "Test email",
+        html: `<p>Hi ${user.name}, this is a test email</p>`,
```

Comment:
Again, meaningful message and find eligible students as recipients.

Reply:
This is finding eligible students `user.email` is the email of respective student